dolphin/alert_manager/trap_receiver.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2018 The OpenSDS Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+**trap receiver**
+"""
+from oslo_log import log
+
+from pysnmp.entity import engine, config
+from pysnmp.carrier.asyncore.dgram import udp
+from pysnmp.entity.rfc3413 import ntfrcv
+from pysnmp.proto.api import v2c
+from pysnmp.smi import builder, view, rfc1902, error
+
+from dolphin.alert_manager import constants
+
+LOG = log.getLogger(__name__)
+
+# Currently static mib file list is loaded, logic to be changed to load all mib file
+MIB_LOAD_LIST = ['SNMPv2-MIB','IF_MIB']
+
+class TrapReceiver(object):
+
+    """Trap receiver functions"""
+    def __init__(self, mibViewController=None, snmpEngine=None):
+        self.mibViewController = mibViewController
+        self.snmpEngine = snmpEngine
+
+    # Loads the list of mib files provided
+    def _mib_builder(self):
+        mibBuilder = builder.MibBuilder()
+        try:
+            self.mibViewController = view.MibViewController(mibBuilder)
+
+            mib_list = MIB_LOAD_LIST
+            if (len(mib_list) > 0):
+                mibBuilder.loadModules_new(*mib_list)
+
+        except error.MibNotFoundError as excep:
+            LOG.exception("Mib load failed")
+
+    # Configures the transport parameters for the snmp engine
+    def _add_transport(self):
+        try:
+            config.addTransport(
+                self.snmpEngine,
+                udp.domainName,
+                udp.UdpTransport().openServerMode((constants.DEF_TRAP_RECV_ADDR, int(constants.DEF_TRAP_RECV_PORT)))
+            )
+        except Exception as e:
```

Comment:
In general, its good to log the error value/traceback.print_exc for better debugging..You can choose the level